lcmgen/emit_java.c

```diff
@@ -528,37 +605,41 @@ int emit_java(lcmgen_t *lcm)
         emit(0, "import java.util.*;");
         emit(0, "import lcm.lcm.*;");
         emit(0, " ");
-        emit(0, "public final class %s %s", lr->structname->shortname,
+        emit_comment(f, 0, structure->comment);
+        emit(0, "public final class %s %s", structure->structname->shortname,
              getopt_get_string(lcm->gopt, "jdecl"));
         emit(0, "{");
 
-        for (unsigned int member = 0; member < g_ptr_array_size(lr->members); member++) {
-            lcm_member_t *lm = (lcm_member_t *) g_ptr_array_index(lr->members, member);
+        for (unsigned int member_index = 0; member_index < g_ptr_array_size(structure->members);
+             member_index++) {
+            lcm_member_t *member =
+                (lcm_member_t *) g_ptr_array_index(structure->members, member_index);
             primitive_info_t *pinfo =
-                (primitive_info_t *) g_hash_table_lookup(type_table, lm->type->lctypename);
+                (primitive_info_t *) g_hash_table_lookup(type_table, member->type->lctypename);
 
+            emit_member_comment(f, 1, member->comment, member);
             emit_start(1, "public ");
 
             if (pinfo == NULL) {
-                emit_continue("%s", make_fqn(lcm, lm->type->lctypename));
+                emit_continue("%s", make_fqn(lcm, member->type->lctypename));
             } else {
                 emit_continue("%s", pinfo->storage);
             }
 
-            emit_continue(" %s", lm->membername);
-            for (unsigned int i = 0; i < g_ptr_array_size(lm->dimensions); i++)
+            emit_continue(" %s", member->membername);
+            for (unsigned int i = 0; i < g_ptr_array_size(member->dimensions); i++)
                 emit_continue("[]");
-            emit_end(";");
+            emit_end(";\n");
```

Comment:
This change does add an unnecessary newline in between the constructor and the last member, but I can't think of a good alternative. It could be removed from here and `emit(0, "");` could be added on L124 before the `emit(indent, "/**");` in `emit_member_comment`, but then that just adds an unnecessary extra newline in between the `class` line and its members.

I'm inclined to just merge as-is since this PR adds a lot of value; we can always open a follow-on PR if we want to change it and can think of a better solution.